### PR TITLE
Development

### PR DESCRIPTION
**Pull request by mossbeard**
@current version 26.10.2018

### CODE REVIEW

File: src/Properties/PublishProfiles/FolderProfile.pubxml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--
+https://go.microsoft.com/fwlink/?LinkID=208121. 
+-->
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <PublishProtocol>FileSystem</PublishProtocol>
+    <Configuration>Release</Configuration>
+    <Platform>Any CPU</Platform>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (konne) [inline src/Properties/PublishProfiles/FolderProfile.pubxml]: why netcore2.0??

File: src/SerConAai.csproj [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
     <RootNamespace>Ser.ConAai</RootNamespace>
     <Authors>Konrad Mattheis und Martin Berthold</Authors>
     <Copyright>Copyright (c) 2018</Copyright>
     <Company>senseexcel</Company>
     <Configuration>Release</Configuration>
-    <Version>$(GitVersion_NuGetVersion)</Version>
+    <Version>$(GitVersion_NuGetVersion)</Version>
```
Comment (konne) [inline src/SerConAai.csproj]: please remove space

File: src/SerEvaluator.cs [hunk 3]
```diff
@@ -492,7 +476,7 @@ private void WaitForDataLoad(ActiveTask task, UserParameter parameter, Cancellat
             {
                 while (true)
                 {
-                    Thread.Sleep(500);
+                    Thread.Sleep(1000);
```
Comment (konne) [inline src/SerEvaluator.cs]: why you have to wait 1 sec.

File: src/SerEvaluator.cs [hunk 4]
```diff
@@ -630,8 +614,8 @@ private void FindTemplatePaths(UserParameter parameter, SerConfig config, string
                 }
                 else if (templateUri.Scheme.ToLowerInvariant() == "lib")
                 {
-                    var connections = GetConnections(onDemandConfig.Connection.ServerUri, parameter.AppId, cookie);
-                    dynamic libResult = connections?.FirstOrDefault(n => n["qName"]?.ToString()?.ToLowerInvariant() == result.Item2) ?? null;
+                    var connections = parameter.SocketConnection.GetConnectionsAsync().Result;
```
Comment (konne) [inline src/SerEvaluator.cs]: don't use this Sync Calls for .Result -> Task -> ContuineWith or Async / Await

File: src/SerEvaluator.cs [hunk 5]
```diff
@@ -711,6 +679,10 @@ private SerConfig GetNewJson(UserParameter parameter, string userJson, string wo
             if (ondemandObject != null)
                 parameter.OnDemand = serConfig["onDemand"]?.ToObject<bool>() ?? false;
 
+            var configConnection = JObject.Parse(JsonConvert.SerializeObject(mainConnection, Formatting.Indented));
+            configConnection["credentials"]["cert"] = null;
```
Comment (konne) [inline src/SerEvaluator.cs]: why you don't use the dynamic way. configConnection.credential.cert ??

File: src/TaskManager.cs [hunk 6]
```diff
@@ -222,7 +220,44 @@ public ActiveTask CreateTask(UserParameter parameter)
...
         }
 
-        public SessionInfo GetSession(SerConnection connection, ActiveTask task)
+        public IDoc GetSessionAppConnection(Uri uri, Cookie cookie, string appId)
+        {
+            try
+            {
+                var url = ServerUtils.MakeWebSocketFromHttp(uri);
+                var connId = Guid.NewGuid().ToString();
+                url = $"{url}/app/engineData/identity/{connId}";
```
Comment (konne) [inline src/TaskManager.cs]: why you open first global and than with global.OpenDocAsync the app.
better is to direct open /app/{APPID}/identity/{connID}

File: src/Properties/PublishProfiles/FolderProfile.pubxml (hunk 1 above)
Comment (mossbeard) [reply src/Properties/PublishProfiles/FolderProfile.pubxml]: Is solved

File: src/SerConAai.csproj (hunk 2 above)
Comment (mossbeard) [reply src/SerConAai.csproj]: Is sovled

File: src/SerEvaluator.cs (hunk 4 above)
Comment (mossbeard) [reply src/SerEvaluator.cs]: Is Implemented

File: src/SerEvaluator.cs (hunk 5 above)
Comment (mossbeard) [reply src/SerEvaluator.cs]: Is Implemented

File: src/SerEvaluator.cs (hunk 3 above)
Comment (mossbeard) [reply src/SerEvaluator.cs]: @konne I need this time for the separate case to load a script in one app.
Here I have to wait until Qlik has completed the Dataload.
Then build the report.
For the standard case, is this code line not significant.

Comment (konne) [reply src/SerEvaluator.cs]: @mossbeard thx. please add a comment to the code for the future

File: src/TaskManager.cs (hunk 6 above)
Comment (konne) [reply src/TaskManager.cs]: @mossbeard please give me here feedback

**Pull request merged**